pkg/agent/multicast/vif_allocator.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicast
+
+import (
+	"container/list"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/vishvananda/netlink"
+	"k8s.io/klog/v2"
+)
+
+// VIF stands for virtual interface for multicast in kernel
+// and is used for adding/deleting static multicast entry in the mfcctl
+// https://github.com/torvalds/linux/blob/master/include/uapi/linux/mroute.h#L80
+// VIFAllocator is used for saving vif <-> interface mapping, also
+// allocate and deallocate VIF because VIF is limited resource(only 32)
```

Comment:
it also allocates and deallocates VIFs

VIFs are limited resources (at most 32)